shared/macros-bash.jinja

```diff
@@ -583,3 +583,16 @@ EOF
 {{% macro bash_deregexify_banner_backslash(banner_var_name) -%}}
 {{{ banner_var_name }}}=$(echo "${{{ banner_var_name }}}" | sed 's/\\//g')
 {{%- endmacro %}}
+
+{{% macro bash_ini_file_set(filename, section, key, value) -%}}
+# Try find '[{{{ section }}}]' and '{{{ key }}}' in '{{{ filename }}}', if it exists, set
+# to '{{{ value }}}', if it isn't here, add it, if '[{{{ section }}}]' doesn't exist, add it there
+if grep -qzosP '[[:space:]]*\[{{{ section }}}]([^\n\[]*\n+)+?[[:space:]]*{{{ key }}}' '{{{ filename }}}'; then
+    sed -i 's/{{{ key }}}[^(\n)]*/{{{ key }}}={{{ value }}}/' '{{{ filename }}}'
+elif grep -qs '[[:space:]]*\[{{{ section }}}]' '{{{ filename }}}'; then
+    sed -i '/[[:space:]]*\[{{{ section }}}]/a {{{ key }}}={{{ value }}}' '{{{ filename }}}'
+else
+    mkdir -p {{{ "/".join(filename.split("/")[:-1]) }}}
```

Comment:
The OVAL evaluates as not applicable if `gdm` is not installed, so this case is covered in case of `oscap ... --remediate`. Generally, that can be said about almost every config file that we remediate. Your point is legitimate, but if we decide to enforce this behavior, let it be a project-wide coordinated effort. The worst thing is a mix of approaches in similar remediations, which is inconsistent, and provides poor guidance to inexperienced content authors that learn by example.

Reply:
oscap shouldn't remediate if not applicable and remediating something that isn't installed is pointless.

Reply:
oscap won't execute this remediation if `gdm` is not installed.

Reply:
Question: Why are we making directories? Seems like we shouldn't be doing this a part of remediations. Are there remediations that need to have a directory created that isn't provided by a rpm?

Reply:
That's a good point, we can probably abort the remediation if the directory doesn't exist.

Reply:
Done, non-applicability is assumed if the parent folder of the config file doesn't exist.